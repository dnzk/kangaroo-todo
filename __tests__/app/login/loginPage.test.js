import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import LoginPage from "../../../src/app/login/page"

describe('LoginPage', () => {
  it('renders correctly', () => {
    const { container } = render(<LoginPage />)
    expect(container).toMatchSnapshot()
  })

  it('renders all elements', () => {
    render(<LoginPage />)

    const logo = screen.getByAltText('Kangaroo Todo')
    expect(logo).toBeInTheDocument()

    const username = screen.getByPlaceholderText('Username')
    expect(username).toBeInTheDocument()

    const password = screen.getByPlaceholderText('Password')
    expect(password).toBeInTheDocument()

    const loginButton = screen.getByText('Log In')
    expect(loginButton).toBeInTheDocument()
  })

  it('disables button when inputs are empty', () => {
    render(<LoginPage />)

    expect(screen.getByText(/log in/i).closest('button')).toBeDisabled()

    fireEvent.change(screen.getByPlaceholderText(/username/i), {
      target: {
        value: 'asdf'
      }
    })

    fireEvent.change(screen.getByPlaceholderText(/password/i), {
      target: {
        value: 'asdf'
      }
    })

    expect(screen.getByText(/log in/i).closest('button')).not.toBeDisabled()
  })
})
