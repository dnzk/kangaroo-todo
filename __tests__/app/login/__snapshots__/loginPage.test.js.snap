// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="login"
    >
      <img
        alt="Kangaroo Todo"
        data-nimg="1"
        decoding="async"
        height="181"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="181"
      />
      <form>
        <input
          class="className base "
          placeholder="Username"
          style="margin-bottom: 25px;"
          type="text"
        />
        <input
          class="className base "
          placeholder="Password"
          style="margin-bottom: 50px;"
          type="text"
        />
        <button
          class="className primary"
          disabled=""
        >
          Log In
        </button>
      </form>
    </div>
  </div>
</div>
`;
